src/EditorFeatures/Core/EmbeddedLanguages/IEmbeddedLanguageEditorFeatures.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Features.EmbeddedLanguages;
+
+namespace Microsoft.CodeAnalysis.Editor.EmbeddedLanguages
+{
+    /// <summary>
+    /// Services related to a specific embedded language.
+    /// </summary>
+    internal interface IEmbeddedLanguageEditorFeatures : IEmbeddedLanguageFeatures
+    {
+        /// <summary>
+        /// A optional brace matcher that can match braces in an embedded language string.
```

Comment:
>optional [](start = 14, length = 8)

Why "optional"?
Given that brace matching is the only feature of an embedded language (at this layer), why would you have one that doesn't do brace matching? Is it for future-proofing?

Reply:
Yes.  There's no necessity for an embedded language to support *all* these capabilities.  For example, 'json' supports code-fixes, while 'regex' does not (currently at least).  These are ways for an embeded language to say: "i support this!".  But it's not a requirement that the language *must* support these things.